packages/forklift-console-plugin/src/modules/Plans/actions/PlanActionsDropdownItems.tsx

```diff
@@ -67,44 +53,79 @@ export const PlanActionsDropdownItems = ({ data }: PlanActionsDropdownItemsProps
     showModal(<PlanDeleteModal resource={plan} model={PlanModel} />);
   };
 
-  return [
-    <DropdownItemLink value={0} key="EditPlan" href={planURL} description={t('Edit Plan')} />,
+  const startActionDescription = useMemo(() => {
+    if (isPlanValidating) {
+      return t('The plan is being validated');
+    }
+
+    switch (status) {
+      case PlanSummaryStatus.Archived:
+        return t('Archived plans cannot be started');
+      case PlanSummaryStatus.Complete:
+        return t('All VMs were migrated');
+      case PlanSummaryStatus.Running:
+        return t('The plan is currently in progress');
+      case PlanSummaryStatus.CannotStart:
+        return t('The plan cannot be started');
+    }
+  }, [status, isPlanValidating]);
```

Comment:
i think this can be extracted to a util file also duplicateActionDescription below can be extracted